### DISCUSSION TITLE
Trigger Azure ML Endpoint already deployed

### DISCUSSION THREAD

**Discussion by tensor365** [Azure/airflow-provider-azure-machinelearning#11] [Q&A]
Hi, 

I would like to know if it is possible to trigger an Azure ML Endpoint already deployed from airflow. I don't want to build all the components, the job and deploy it on azure. I would like just to trigger the endpoint. If it's not possible, we'll still use requests combined to MSIAuthentification. But this connector will be a good opportunity to factorize the connection in our dags. 

Thank you

**Comment by saipulla** [comment]
Hello, @tensor365.

Thanks for the question. Being able to trigger an endpoint from Airflow is in the plans. We will work on it, once we have resources/cycles. 

For the short term, there is an easy work around: 
1. To invoke/trigger an endpoint, follow the examples like [this one](https://github.com/Azure/azureml-examples/blob/main/sdk/python/endpoints/batch/text-summarization-batch.ipynb). 
2. Extend the example operation from [here](https://github.com/Azure/airflow-provider-azure-machinelearning/blob/351e0ec541211bd69de96d1256f3b27c0520c659/airflow_provider_azure_machinelearning/example_dags/AML_PyEnv.py#L215) to invoke a named endpoint with data.

That would allow you to employ PythonVirtualenvOperator from Airflow to trigger the endpoints.

**Reply by tensor365** [reply]
Hi saipula,

If i work on a fork to add the operator Can it help ?

**Reply by saipulla** [reply]
Sure you are welcomed to work on it via fork. You are also welcomed to just propose a PR on main branch on this fork., as long as the changes passes the existing PR gates and in good quality, I am sure other users will also appreciate your contribution.
Sorry for the late response.

**Comment by tensor365** [comment]
No problem for timing. I added in job operators, a new operator but it does not rely on sdk v2. Indeed with sdk2 it seems that we have to rebuild the entire job via the sdk or external yaml. It's quite painful if we just want to submit same job with différents parameters. So for the moment we just call the endpoint by using token auth and sdk2 to check job status. Will it be enough to pass your quality test ?

**Reply by tensor365** [reply]
My preceding message about sdk1 and 2 IS obsolète you just published a New feature to use pipeline in batchendpoint. I will just add an invoke operator so. I'll try to submit to you before de the 15th

**Reply by tensor365** [reply]
I finished the dev I'll submit the code to you next week
